From 962e6f3bf572123e858fcbc34bb63af307de0a9c Mon Sep 17 00:00:00 2001
From: sisah <jirielbunda@seznam.cz>
Date: Fri, 10 Mar 2023 14:50:24 -0500
Subject: [PATCH] hardcode
 0014-settingswindow-save-user-settings-file-when-ok-is-pr.patch

---
 apps/openmw/main.cpp                 | 3 +++
 apps/openmw/mwgui/settingswindow.cpp | 5 +++++
 2 files changed, 8 insertions(+)

diff --git a/apps/openmw/main.cpp b/apps/openmw/main.cpp
index 1f1c114e717..91bc0a2640c 100644
--- a/apps/openmw/main.cpp
+++ b/apps/openmw/main.cpp
@@ -216,6 +216,7 @@ namespace
         }
     };
 }
+Files::ConfigurationManager *g_cfgMgr;
 
 int runApplication(int argc, char* argv[])
 {
@@ -227,6 +228,7 @@ int runApplication(int argc, char* argv[])
 
     osg::setNotifyHandler(new OSGLogHandler());
     Files::ConfigurationManager cfgMgr;
+    g_cfgMgr = &cfgMgr;
     std::unique_ptr<OMW::Engine> engine = std::make_unique<OMW::Engine>(cfgMgr);
 
     // Android navmeshtool hack
diff --git a/apps/openmw/mwgui/settingswindow.cpp b/apps/openmw/mwgui/settingswindow.cpp
index 6c6a34595e0..e3c13ffc5e7 100644
--- a/apps/openmw/mwgui/settingswindow.cpp
+++ b/apps/openmw/mwgui/settingswindow.cpp
@@ -27,6 +27,7 @@
 #include <components/resource/resourcesystem.hpp>
 #include <components/resource/scenemanager.hpp>
 #include <components/sceneutil/lightmanager.hpp>
+#include <components/files/configurationmanager.hpp>
 #include <components/settings/values.hpp>
 #include <components/vfs/manager.hpp>
 #include <components/vfs/recursivedirectoryiterator.hpp>
@@ -142,6 +143,7 @@ namespace
             box->setIndexSelected(MyGUI::ITEM_NONE);
     }
 }
+extern Files::ConfigurationManager *g_cfgMgr;
 
 namespace MWGui
 {
@@ -451,6 +453,8 @@ namespace MWGui
 
     void SettingsWindow::onOkButtonClicked(MyGUI::Widget* _sender)
     {
+        const std::string settingspath = (g_cfgMgr->getUserConfigPath() / "settings.cfg").string();
+        Settings::Manager::saveUser(settingspath);
         setVisible(false);
     }
 
